File: JEONGHO/codes/2_function.cpp
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// 도서 정보 구조체 정의
typedef struct {
    char title[100];
    char author[100];
    char publisher[100];
} Book;

// 데이터베이스 저장 함수 (단순화)
void saveToDatabase(Book book) {
    // 여기서 MySQL 또는 다른 DB에 연결하여 데이터를 저장하는 코드가 필요
    printf("도서 정보 저장: %s, %s, %s\n", book.title, book.author, book.publisher);
}

// 도서 유사성 검토 함수 (단순화)
int compareBooks(Book book1, Book book2) {
    return strcmp(book1.title, book2.title) == 0 &&
           strcmp(book1.author, book2.author) == 0 &&
           strcmp(book1.publisher, book2.publisher) == 0;
}

// 네이버 도서 API 사용하여 분류 정보 가져오기 (단순화)
void getCategoryFromNaver(Book book) {
    // 네이버 도서 API 연동 코드 필요
    printf("네이버 API 호출: %s\n", book.title);
    printf("분류 정보: 과학\n"); // 예시로 '과학' 분류 출력
}

// 신간 도서 목록 파일 읽기
int loadNewBooks(const char *filename, Book *books, int maxBooks) {
    FILE *file = fopen(filename, "r");
    if (file == NULL) {
        perror("파일 열기 실패");
        return 0;
    }
    
    int count = 0;
    while (count < maxBooks && fscanf(file, "%[^,], %[^,], %[^\n]\n", books[count].title, books[count].author, books[count].publisher) == 3) {
        count++;
    }
    fclose(file);
    return count;
}

// 단계 1: 카메라 인식을 통해 타겟 도서 정보 획득 (단순화)
Book detectBookFromCamera() {
    Book book;
    printf("카메라로부터 도서 인식 중...\n");
    // 여기서 실제 객체 인식 코드가 필요
    strcpy(book.title, "인식된 책 제목");
    strcpy(book.author, "인식된 저자명");
    strcpy(book.publisher, "인식된 출판사명");
    return book;
}

int main() {
    // 0단계: 신간 도서 구입 목록 파일 불러오기
    Book newBooks[100];
    int newBookCount = loadNewBooks("new_books.txt", newBooks, 100);
    printf("신간 도서 %d권 불러옴.\n", newBookCount);

    // 1단계: 카메라로 타겟 도서 인식
    Book targetBook = detectBookFromCamera();
    
    // 2단계 & 3단계: 유사성 검토
    int isMatchFound = 0;
    for (int i = 0; i < newBookCount; i++) {
        if (compareBooks(newBooks[i], targetBook)) {
            printf("타겟 도서와 신간 목록 일치: %s\n", targetBook.title);
            isMatchFound = 1;
            break;
        }
    }

    // 4단계: 타겟 도서 정보 저장
    if (!isMatchFound) {
        printf("타겟 도서가 신간 목록에 없습니다. 정보 확정 후 저장합니다.\n");
    }
    saveToDatabase(targetBook);

    // 5단계: 필요시 네이버 도서 API로 분류 정보 가져오기
    getCategoryFromNaver(targetBook);

    return 0;
}

File: JEONGHO/codes/2_function.py
# 신간 도서 기능 추가
# 0.문제 내용/입력/출력/
# 1.조건 파악
# 2.프로그램(시스템) 구성도
# 3.사용한 개념/자료구조
# 4.프로그램 특징
# 5.코드

# 0.문제 내용/입력/출력/
- 문제 내용: 신간 도서 추가 기능
- 입력: 신간 도서 구입 목록(txt / xlsx), 책장에 신간 도서 배치
- 출력: 신간 타겟 도서 



# 1.조건 파악
    - 수직 텍스트 책 제목 미포함
    
# 2.프로그램(시스템) 구성도
    0) 신간 도서 구입 목록 시스템에 입력
        - 도서명/저자/출판사/[책장/책장 내 인덱스] = [분류 기호표로 치환 가능?]
    1) 도서관에 새로 도착
    2) 신간 도서를 책장에 배치
    3) 해당 책장의 카메라가 신간 도서 객체 인식
    4) 타겟 도서 책 제목/저자명/출판사 추출
    4) (0단계에서 시스템에 입력된 신간 도서 구입 목록과)유사성 검토
    5) 타겟 도서의 정확한 정보 확정
    6) 도서관 DB에 타겟 도서 데이터 등록
    7) 분류 관련 정보는 네이버 도서 api 활용 분류 분야(ex. 과학/인문/에세이 등 파악)

# 3.사용한 개념/자료구조
구매 목록, 
# 4.프로그램 특징
# 5.코드

